src/plugins/intel_gpu/src/plugin/plugin.cpp

```diff
@@ -160,6 +160,15 @@ Plugin::Plugin() {
     m_compiled_model_runtime_properties["OV_VERSION"] = ov_version.buildNumber;
 }
 
+Plugin::~Plugin() {
+#ifdef ENABLE_ONEDNN_FOR_GPU
+    // To prevent hanging during oneDNN's primitive cache desctruction,
+    // trigger earlier cache cleanup by setting its capacity to 0.
+    // Related ticket: 106154.
+    dnnl::set_primitive_cache_capacity(0);
```

Comment:
what if 2 Core objects are created and GPU device is initialized?

Reply:
@ilya-lavrenov good point, thanks for the comment
I have updated the logic a little and added restoring of original cache capacity right after cache cleanup. In case of multiple ov::Core instances the cache cleanup may affect performance by evicting one plugin's entries while calling another plugin destructor. But at least for now, the cache capacity will be restored after cleaning it up. Hope it isn't a common usage scenario. I also updated a ticket for oneDNN team (please see OV's 106154 related tickets)